File: solutions/julia/grains/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
    if square < 1 || square > 64
        throw(DomainError(""))
    else
        return UInt128(2)^(square - 1)
    end
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    if square < 1 || square > 64
        throw(DomainError(""))
    else
        return sum([on_square(n) for n in 1:square])
    end
end

File: solutions/cpp/grains/grains.cpp
#include "grains.h"

#include <cstdint>
#include <cmath>

namespace grains {

    auto square(uint64_t n) -> uint64_t
    {
        return std::pow(2, n - 1);
    }

    auto total() -> uint64_t
    {
        uint64_t t = 0;

        for(int i = 1; i <= 64; i++)
            t += square(i);

        return t;
    }

}  // namespace grains
